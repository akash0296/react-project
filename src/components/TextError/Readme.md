Text Error

```jsx inside Markdown
import mockData from './__test__/TextError.mock.json'
;<TextError error={mockData.error} />
```
