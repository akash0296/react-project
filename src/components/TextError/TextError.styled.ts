import styled from 'styled-components'

const ErrorText = styled.div`
	display: flex;
	align-items: center;
	svg {
		width: 13px;
		fill: ${(props: any) => props.theme.colors.PERSIAN_RED};
	}
	p {
		color: ${(props: any) => props.theme.colors.PERSIAN_RED};
		font-size: 14px;
		margin-left: 5px;
	}
`

ErrorText.displayName = 'ErrorTextBlock'

export { ErrorText }
