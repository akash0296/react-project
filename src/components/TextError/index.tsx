import React from 'react'

import { Error } from 'assets/icons'
import { ErrorText } from './TextError.styled'

type Props = {
	/**
	 * error message
	 */
	error?: string
}

const TextError = ({ error }: Props) => {
	if (!error) {
		return null
	}

	return (
		<ErrorText>
			<Error />
			<p>{error}</p>
		</ErrorText>
	)
}

export default TextError
