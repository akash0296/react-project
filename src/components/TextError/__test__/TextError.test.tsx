import React from 'react'
import { shallow } from 'enzyme'

import TextError from '../index'
import mockData from './TextError.mock.json'

describe('Text Error component tests', () => {
	let TextErrorComp: any = null
	const handleChangeStub = jest.fn()
	beforeEach(() => {
		TextErrorComp = shallow(<TextError error={mockData.error} />)
	})

	it('should render TextError Component', () => {
		expect(TextErrorComp.find('ErrorTextBlock').length).toEqual(1)
	})

	it('matches the snaphot', () => {
		expect(TextErrorComp).toMatchSnapshot()
	})
})
