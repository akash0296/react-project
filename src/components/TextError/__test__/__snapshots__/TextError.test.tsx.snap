// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Error component tests matches the snaphot 1`] = `
<ErrorTextBlock>
  <ForwardRef(SvgError) />
  <p>
    Test Error Text
  </p>
</ErrorTextBlock>
`;
